float64 E_pos
float64 E_theta